FirebaseDynamicLinks/Sources/FIRDynamicLinks.m

```diff
@@ -509,7 +509,20 @@ - (BOOL)handleIncomingCustomSchemeDeepLink:(NSURL *)url {
 }
 
 - (void)passRetrievedDynamicLinkToApplication:(NSURL *)url {
-  [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
+  id<UIApplicationDelegate> applicationDelegate = [UIApplication sharedApplication].delegate;
+  if (applicationDelegate &&
+      [applicationDelegate respondsToSelector:@selector(application:openURL:options:)]) {
+    // pass url directly to application delegate to avoid hop into
+    // iOS handling of the universal links
+    if (@available(iOS 9.0, *)) {
+      [applicationDelegate application:[UIApplication sharedApplication] openURL:url options:@{}];
+      return;
+    }
+  }
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  [[UIApplication sharedApplication] openURL:url];
```

Comment:
Is it possible to use an un-deprecatedAPI here?

Reply:
Done